packages/zoltan2/src/algorithms/color/Zoltan2_AlgSerialGreedy.hpp

```diff
@@ -130,8 +130,8 @@ class AlgSerialGreedy : public Algorithm<Adapter>
     // Find max degree, since (max degree)+1 is an upper bound.
     lno_t maxDegree = 0; 
     for (size_t i=0; i<nVtx; i++){
-      if (offsets[i+1]-offsets[i] > maxDegree)
-        maxDegree = offsets[i+1]-offsets[i];
+      if (offsets[i+1]-offsets[i] > static_cast<offset_t>(maxDegree))
+        maxDegree = static_cast<lno_t>(offsets[i+1]-offsets[i]);
```

Comment:
Why not change the type of maxDegree to offset_t rather than do the casts?
Do the Teuchos::Array constructors require lno_t (or not work with size_t)?